Engine/AlgorithmManager.cs

```diff
@@ -728,10 +705,8 @@ public void Run(AlgorithmNodePacket job, IAlgorithm algorithm, ISynchronizer syn
             SetStatus(AlgorithmStatus.Completed);
 
             //Take final samples:
-            results.SampleRange(algorithm.GetChartUpdates());
-            results.SampleEquity(_previousTime, Math.Round(algorithm.Portfolio.TotalPortfolioValue, 4));
-            SampleBenchmark(results, backtestMode ? _previousTime.Date : _previousTime);
-            SamplePerformance(results, _previousTime.Date, force:true);
+            results.Sample(previousTime, force: true);
```

Comment:
Why use `previousTime` here? Since we are outside the streaming loop at this stage, wouldn't both be equals? Maybe we can join `time` in the loop with this `previousTime` outside the loop, into a single global `time`?